### ISSUE TITLE
Mastodon link is interpreted as email address

### ISSUE THREAD

**Issue by almereyda** [lycheeverse/lychee#1462]
When running Lychee on [degrowth.net/organisations/instituto-resiliencia/index.html](https://degrowth.net/organisations/instituto-resiliencia/index.html), it reports, that the email address cannot be reached:

> #### Errors in public/organisations/instituto-resiliencia/index.html
>
>    [ERR] [instituto_resiliencia@15-15-15.social](https://github.com/degrowth/idn-website/pull/instituto_resiliencia@15-15-15.social) | Failed: Unreachable mail address: [instituto_resiliencia@15-15-15.social](mailto:instituto_resiliencia@15-15-15.social): Invalid: Email doesn't exist or is syntactically incorrect
>
> https://github.com/degrowth/idn-website/pull/23

This is a special case, since the string that is recognised is only used as the label for a hyperlink to `https://15-15-15.social/@instituto_resiliencia`. The link clearly does not point at a `mailto:` URI.

This is because of a design choice made in how Mastodon addresses are displayed. The @ is shown directly before of the string that is recognised as an email address, but not part of the link anchor.

![grafik](https://github.com/lycheeverse/lychee/assets/1645308/e07970b6-9e10-4ac2-b75d-f6c90fa4c408)

    <p>Mastodon: @<a href=[https://15-15-15.social/@instituto_resiliencia](view-source:https://15-15-15.social/@instituto_resiliencia) target=_blank>instituto_resiliencia@15-15-15.social</a><br>Twitter: @<a href=[https://twitter.com/ins_resiliencia](view-source:https://twitter.com/ins_resiliencia) target=_blank>ins_resiliencia</a></p>

The HTML in action:

> <p>Mastodon:&nbsp;@<a href="https://15-15-15.social/@instituto_resiliencia" target="_blank">instituto_resiliencia@15-15-15.social</a><br>Twitter: @<a href="https://twitter.com/ins_resiliencia" target="_blank">ins_resiliencia</a></p>

It appears `check-if-email-exists`, of which an outdated fork exists at [github.com/lycheeverse/check-if-email-exists](https://github.com/lycheeverse/check-if-email-exists), which has fallen back considerably behind [github.com/reacherhq/check-if-email-exists](https://github.com/reacherhq/check-if-email-exists), does not recognise that the link points some place else. Eventually by grepping the whole file for a regular expression matching an email pattern, not considering the semantics of the HTML.

What would be the best way to proceed from here?

**Comment by almereyda** [comment]
One way to circumvent this is to set `include_mail` to `false`.

But can the programmatic regression of not being able to distinguish HTML links to URLs from links to `mailto:` URIs be remediated?

**Comment by mre** [comment]
I think the issue runs even deeper than the incorrect detection of email addresses.

lychee incorrectly parses both the `href` attribute and the link text as separate URLs, leading to false positives and potential misinterpretation of links.

## Reproduction Steps
1. Create a test file:
```bash
echo '<a href="https://example.com/real">https://example.com/fake</a>' > test.html
```

2. Run lychee:
```bash
lychee -v --dump test.html
```

Expected: Only `https://example.com/real` is detected
Actual: Both `https://example.com/real` and `https://example.com/fake` are detected

## Proposed Fix
Update lychee to only parse the `href` attribute for URLs when examining HTML content, ignoring the link text.

## Test Cases
1. HTML link with different URL in text:
   ```bash
   echo '<a href="https://example.com/real">https://example.com/fake</a>' | lychee -
   ```
2. Mastodon-style link:
   ```bash
   echo '<a href="https://mastodon.social/@user">user@mastodon.social</a>' | lychee -
   ```

## Impact
This fix would resolve both the original Mastodon link issue and prevent false positives from link text that may contain URL-like strings.

**Comment by mre** [comment]
Perfect. Let's catch up tomorrow. You can reach me on here or via Mastodon. mre@mastodon.social

**Comment by mre** [comment]
1. Yes, these should be included, but I'd update the tests instead of culling them because of point 2. 
2. Yes. It sounds like it's a good idea to include those links when `verbatim` is set. Maybe we can keep the tests but set them up so that they test verbatim mode?

**Comment by mre** [comment]
Fixed by @markogalevski. This fix first appeared in [lychee-lib-v0.17.0](https://github.com/lycheeverse/lychee/releases/tag/lychee-lib-v0.17.0).
